pkg/platform/packet/packet.go

```diff
@@ -234,24 +234,37 @@ func createTerraformConfigFile(cfg *config, terraformPath string) error {
 }
 
 // terraformSmartApply applies cluster configuration.
-func (c *config) terraformSmartApply(ex *terraform.Executor, dnsProvider dns.DNSProvider) error {
+func (c *config) terraformSmartApply(ex *terraform.Executor, dnsProvider string, dnsZone string) error {
 	// If the provider isn't manual, apply everything in a single step.
 	if dnsProvider != dns.DNSManual {
 		return ex.Apply()
 	}
 
 	arguments := []string{"apply", "-auto-approve"}
 
-	// Get DNS entries (it forces the creation of the controller nodes).
-	arguments = append(arguments, fmt.Sprintf("-target=module.packet-%s.null_resource.dns_entries", c.ClusterName))
+	// Create controllers. We need the controllers' IP addresses before we can
+	// apply the 'dns' module.
+	arguments = append(arguments, fmt.Sprintf("-target=module.packet-%s.packet_device.controllers", c.ClusterName))
+	if err := ex.Execute(arguments...); err != nil {
+		return errors.Wrap(err, "creating controllers")
+	}
 
-	// Create controller
+	// Apply 'dns' module.
+	arguments = append(arguments, "-target=module.dns")
 	if err := ex.Execute(arguments...); err != nil {
-		return errors.Wrap(err, "failed executing Terraform")
+		return errors.Wrap(err, "applying 'dns' module")
```

Comment:
Why?

Reply:
https://github.com/kinvolk/lokomotive/issues/59

Reply:
But there is a lot of adjacent code which uses `errors.Wrap`. I'd rather not introduce inconsistencies. If https://github.com/kinvolk/lokomotive/pull/388/commits/75504147e2678699ac640b38abd9a2743ce09b1b is merged before this PR I can rebase and make the change, otherwise I'd like to keep things consistent.